osbrain/agent.py

```diff
@@ -193,20 +193,20 @@ class Agent():
         Default agent serialization format.
     transport : str, AgentAddressTransport, default is None
         Default agent transport protocol.
-    socket : dict
+    _socket : dict
         A dictionary in which the key is the address or the alias and the
         value is the actual socket.
-    address : dict
+    _address : dict
         A dictionary in which the key is the address or the alias and the
         value is the actual address.
-    handler : dict
+    _handler : dict
         A dictionary in which the key is the socket and the values are the
         handlers for each socket.
-    context : zmq.Context
+    _context : zmq.Context
         ZMQ context to create ZMQ socket objects.
-    poller : zmq.Poller
+    _poller : zmq.Poller
         ZMQ poller to wait for incoming data from sockets.
-    poll_timeout : int
+    _poll_timeout : int
         Polling timeout, in milliseconds. After this timeout, if no message
         is received, the agent executes de `idle()` method before going back
         to polling.
```

Comment:
Maybe `keep_alive` and `running` attributes should be hidden too.

Reply:
Maybe `keep_alive` should be changed (it's used in one test, though).

I'm not sure about `running`, though. It's used very frequently to check if the agent is running or not. Maybe we should make the attribute private and add a method to return its value?

Reply:
Yeah, an `.is_running()` method (or something like that) makes a lot of sense.